crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -377,43 +382,9 @@ def upgrade_drbd_cluster(cluster)
         end
         return false if drbd_slave.empty?
 
-        node_api = Api::Node.new drbd_slave
+        return false unless upgrade_first_cluster_node drbd_slave, drbd_master
 
-        save_upgrade_state("Starting the upgrade of node #{drbd_slave}")
-        return false unless node_api.upgrade
-
-        # Explicitly mark node1 as the cluster founder
-        # and adapt DRBD config to the new founder situation.
-        # This shoudl be one time action only (for each cluster).
-        unless Api::Pacemaker.set_node_as_founder drbd_slave
-          save_error_state("Changing the cluster founder to #{drbd_slave} has failed")
-          return false
-        end
-
-        # Remove "pre-upgrade" attribute from node1
-        # We must do it from a node where pacemaker is running
-        master_node_api = Api::Node.new drbd_master
-        return false unless master_node_api.disable_pre_upgrade_attribute_for drbd_slave
-
-        # FIXME: this should be one time action only (for each cluster)
-        return false unless delete_pacemaker_resources drbd_master
-
-        # Execute post-upgrade actions after the node has been upgraded, rebooted
-        # and the existing cluster has been cleaned up by deleting most of resources:
-        # - start pacemaker and sync DRBD devices
-        return false unless node_api.post_upgrade
-        # - initiate the first chef-client run
-        return false unless node_api.join_and_chef
-
-        # migrate routers from drbd_master to recently upgraded drbd_slave
-        return false unless node_api.router_migration
```

Comment:
I am pretty sure we'll still need the method. But I think we should worry about that in a different PR.